### ISSUE TITLE
App wont start on High Sierra after enabling Touchbar

### ISSUE THREAD

**Issue by pteasima** [fifafu/BetterTouchTool#1056]
TLDR; version if you dont want to spend time supporting macos betas: Can you point me to where you store some settings/caches outside of Btt.app? I just want to get a clean start without having to format and reinstall macos. I dont care if I wont be able to use btt for touchbar. Thanks!

#### Description of bug/feature request/question:

When I enable Touchbar and restart BTT, the app never starts. I can see "btt relaunch" in activity monitor running forever. I've attached a sample I took with activity monitor, so you can see the stack trace (it's always the same no matter how long I wait, so you can be confident that it hangs in that particular method). 

This behaviour is consistent (I actually reformatted my whole mac and reinstalled High Sierra for another reason, so it has happened to me twice). Deleting and redownloading the app doesnt help. Restarting the mac doesnt help. If I uncheck btt in accessibility settings and restart btt, the "enable accesibility window appears for a split second, then dissapears and the rest is the same.

Steps to reproduce:
Start btt on fresh install of High Sierra beta 1, enable accessibility, import my profile (attached zip), delete default profile, enable touchbar in "general touchbar settings" (btt icon doesnt appear in touchbar), relaunch btt (never starts as described above).

[btt.json.zip](https://github.com/fifafu/BetterTouchTool/files/1064554/btt.json.zip)
#### Affected input device (e.g. MacBook Trackpad, Magic Mouse/Trackpad, Touch Bar, etc.):

N/A

#### Device information:

Type of Mac: 2016 macbook pro 13 with touchbar
macOS version: 10.13 Beta (17A264c)
BetterTouchTool version: I think its 2.10, its the latest version as of today (9.6.2017)

#### Additional information (e.g. StackTraces, related issues, screenshots, workarounds, etc.):

[bttrelaunchsample.txt](https://github.com/fifafu/BetterTouchTool/files/1064528/bttrelaunchsample.txt)

---

**Note:** Before bug reporting, please make sure you have the latest version of BetterTouchTool and that you have already tried to restart your system :-). If you encounter a crash, please attach a crash log from the macOS Console.app from the "User Diagnostic Reports" section.

**Comment by fifafu** [comment]
Hey,

I'll look into the beta this weekend. There is probably nothing you can do about it for now, but I'll provide a working version this weekend.
If possible please provide the crash reports from the console.app -> user reports, they'll help me identify the issues :-)

**Comment by pteasima** [comment]
Thanks for the quick reply. Sorry, I never noticed theres a crash report. Looks like its the emoji widget.

[BetterTouchTool_2017-06-08-181747_Petrs-MacBook-Pro.crash.zip](https://github.com/fifafu/BetterTouchTool/files/1064599/BetterTouchTool_2017-06-08-181747_Petrs-MacBook-Pro.crash.zip)

Anyway, I found what I needed. I just had to delete `~/Library/Application\ Support/BetterTouchTool` and now I can start the app again (since it doesnt have my profile with the emoji widget). I feel like I shouldnt close this issue yet, so please deal with it as you wish.

**Comment by pteasima** [comment]
fyi after deleting the emoji widget, everything else works just fine. So there's no rush for you to fix anything. Only problem is that now the app crashes as soon as I import a preset which contains an emoji widget (it didnt crash right away before, only didnt relaunch). So people might have to edit their .json presets manually, as I did.

**Comment by fifafu** [comment]
Awesome! Thanks for letting me know. (The emoji widget is using lots of private api, probably something changed there)

I'll add a fix.

**Comment by fifafu** [comment]
should be fixed in v2.21 alpha https://bettertouchtool.net/releases/btt2.21.zip

Maybe I can integrate the native emoji widget in a future version.
